### PR TITLE
Add code for `unit` target in the Makefile.ci

### PR DESCRIPTION
**Pull request by sohankunkerkar**

### CODE REVIEW

**Comment by pmorie** [comment]
/approve

We are going to figure out what's going on here after this merges, but it is meaningful to have these tests running even if there's a little cruft we need to reduce.

**Comment by pmorie** [comment]
@sohankunkerkar looks like the unit tests failed in CI

**Comment by sohankunkerkar** [comment]
> @sohankunkerkar looks like the unit tests failed in CI

it's bound to happen since we changed the location of Dockerfile from `openshift/Dockerfile` to `Dockerfile` and CI's config still using the old change: 
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/kubefed/openshift-kubefed-master.yaml#L9

**Comment by sohankunkerkar** [comment]
@pmorie  PTAL

File: openshift/Makefile.ci [hunk 1]
```diff
@@ -1,7 +1,17 @@
 # Makefile.ci holds make directives called by the CI jobs for this repo.
+TARGET = kubefed
+GOTARGET = sigs.k8s.io/$(TARGET)
```
Comment (abhat) [inline openshift/Makefile.ci:3]: This should be `github.com/openshift/$(TARGET)`

Comment (sohankunkerkar) [reply openshift/Makefile.ci:3]: It expects a value which is defined for `canonical_go_repository` parameter in config.yaml file. The value you're suggesting doesn't pass the ci-operator tests locally.

**Pull request merged**